tests/sentry/api/endpoints/test_project_rule_details.py

```diff
@@ -616,63 +416,33 @@ def test_update_channel_slack(self):
             body=json.dumps({"ok": channels["ok"], "channel": channels["channels"][1]}),
         )
 
-        response = self.client.put(
-            url,
-            data={
-                "name": "#new_channel_name",
-                "actionMatch": "any",
-                "filterMatch": "any",
-                "actions": actions,
-                "conditions": conditions,
-                "frequency": 30,
-            },
-            format="json",
+        payload = {
+            "name": "#new_channel_name",
+            "actionMatch": "any",
+            "filterMatch": "any",
+            "actions": actions,
+            "conditions": conditions,
+            "frequency": 30,
+        }
+        self.get_success_response(
+            self.organization.slug, self.project.slug, self.rule.id, status_code=200, **payload
         )
-
-        assert response.status_code == 200, response.content
-        rule = Rule.objects.get(id=response.data["id"])
-        assert rule.label == "#new_channel_name"
-        assert rule.data["actions"][0]["channel_id"] == "new_channel_id"
+        # TODO(Leander): Figure out a way to just run assert_from_payload)
+        assert self.rule.data["actions"][0]["channel_id"] == "new_channel_id"
```

Comment:
You might be able to do [`assertIn`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertIn)

Reply:
Unfortunately I couldn't, the function I wrote `assert_rule_from_payload` compares the list of dictionaries passed from the payload, to the list of dictionaries saved on the Rule. We can't do an `in` comparison on the dicts or list of dicts since they might only be a subset of the full dict on the rule 

e.g. payload has `actions: [{name: Leander}]` but save the rule with `data: {actions: [{id: 1, name: Leander}]}`. This should pass the assertion